docs/reference/TensorFlow.md

```diff
@@ -1,405 +1,370 @@
 
-Congratulations! You are now running a machine learning model that can recognize keywords on an Arm Cortex-M4 microcontroller, directly on your K66F.
+    ```
+    xxd -i  tiny_conv.tflite > ../micro_features/model.cc
+    ```
 
-It is easy to change the behavior of our program, but is it difficult to modify the machine learning model itself? The answer is no, and the next section of this guide, [Retrain the machine learning model](https://developer.arm.com/solutions/machine-learning-on-arm/developer-material/how-to-guides/build-arm-cortex-m-voice-assistant-with-google-tensorflow-lite/retrain-the-machine-learning-model), will show you how.
+1. Update `model.cc` so that it is compatible with our code:
 
-# Retrain the machine learning model
+    1. Open the file. The top two lines should look similar to the following code, although the exact variable name and hex values may be different:
 
-The model that we are using for speech recognition was trained on a dataset of one-second spoken commands called the [Speech Commands Dataset](https://ai.googleblog.com/2017/08/launching-speech-commands-dataset.html). The dataset includes examples of the following ten different words:
+        ```
+        const  unsigned  char  g_model[] DATA_ALIGN_ATTRIBUTE = {  
+        0x18, 0x00, 0x00, 0x00, 0x54, 0x46, 0x4c, 0x33, 0x00, 0x00, 0x0e, 0x00,
+        ```
 
-yes, no, up, down, left, right, on, off, stop, go
+    1. Add the `include` from the following snippet and change the variable declaration without changing the hex values:
 
-While the model in this sample was originally trained to recognize “yes” and “no”, the TensorFlow Lite for Microcontrollers source contains scripts that make it easy to retrain the model to classify any other combination of these words.
+        ```
+        #include "tensorflow/lite/micro/examples/micro_speech/micro_features/model.h"  
+        const unsigned char g_tiny_conv_micro_features_model_data[] = {  
+        0x18, 0x00, 0x00, 0x00, 0x54, 0x46, 0x4c, 0x33, 0x00, 0x00, 0x0e, 0x00,
+        ```
 
-We are going to use another pre-trained model to recognize “up” and “down”, instead. If you are interested in the full flow including the training of the model refer to the [Supplementary information: model training](https://developer.arm.com/solutions/machine-learning-on-arm/developer-material/how-to-guides/build-arm-cortex-m-voice-assistant-with-google-tensorflow-lite/supplementary-information-model-training) section of this guide.
+    1. Go to the bottom of the file and find the unsigned int variable:
 
-To build our new ML application we will now follow these steps:
+        ```
+        unsigned int tiny_conv_tflite_len = 18216;
+        ```
 
--   Download a pretrained model that has been trained and frozen using TensorFlow.
+    1. Change the declaration to the following code, but do not change the number assigned to it, even if your number is different from the one in this guide:
```

Comment:
Why? What is it?

Reply:
"Change the variable's name and constness to"?

Reply:
That number is auto-generated by the converter script and the constant is the length of the model parameters. I suggest the following change "Change the declaration to the following code, but do not change the number assigned to it, as it is automatically generated to indicate the size of the model. Even if your number is different from the one in this guide:"

Reply:
The "value"? (as opposed to "number")

Reply:
sounds good let's go with value 👍